lib/doorkeeper/openid_connect/helpers/controller.rb

```diff
@@ -7,7 +7,8 @@ module Controller
         def authenticate_resource_owner!
           super.tap do |owner|
             next unless valid_controller_path?
-            next unless scopes.include?('openid')
+            next unless params[:scope] && params[:scope].include?('openid')
+            raise Errors::OpenidConnectError unless pre_auth.valid?
```

Comment:
We _only_ check that the `PreAuthorization` is valid if openid exists as a scope on params.

In a successful flow, only `scopes` and `client` on the pre_auth are used in this helper.  However, `pre_auth.valid?` will also check that the following are valid
- `response_type`
- `redirect_uri`
- `code_challenge_method`

Are we breaking part of the RFC by checking `pre_auth.valid?`?  Or do we need to be separately checking `pre_auth.client.valid?` and `pre_auth.scopes.valid?`.  My gut feeling is that too much can go wrong with the OpenID flow should the pre_auth not be valid, however, _verify this requirement or lack thereof_.